applications/IgaApplication/custom_modelers/nurbs_geometry_modeler_sbm.cpp

```diff
@@ -0,0 +1,149 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Nicolo' Antonelli
+//                   Andrea Gorgi
+//
+
+// Project includes
+#include "nurbs_geometry_modeler_sbm.h"
+#include "custom_utilities/create_breps_sbm_utilities.h"
+#include "custom_processes/snake_sbm_process.h"
+#include "iga_application_variables.h"
+
+namespace Kratos
+{
+
+    ///@name Stages
+    ///@{
+
+    void NurbsGeometryModelerSbm::SetupGeometryModel(){
+
+        //----------------------------------------------------------------------------------------------------------------
+        KRATOS_INFO_IF("NurbsGeometryModelerSbm", mEchoLevel > 1) << "[NURBS MODELER SBM]:: calling NurbsGeometryModeler" << std::endl;
+        
+        // Call the SetupGeometryModel method of the base class NurbsGeometryModeler
+        NurbsGeometryModeler::SetupGeometryModel();
+    }
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+    void NurbsGeometryModelerSbm::CreateAndAddRegularGrid2D(ModelPart& rModelPart, const Point& A_xyz, const Point& B_xyz,
+        const Point& A_uvw, const Point& B_uvw, const SizeType OrderU, const SizeType OrderV, const SizeType NumKnotSpansU, 
+        const SizeType NumKnotSpansV, const bool AddSurfaceToModelPart)
```

Comment:
```suggestion
    void NurbsGeometryModelerSbm::CreateAndAddRegularGrid2D(
        ModelPart& rModelPart,
        const Point& rAxyz,
        const Point& rBxyz,
        const Point& rAuvw,
        const Point& rBuvw,
        const SizeType OrderU,
        const SizeType OrderV,
        const SizeType NumKnotSpansU, 
        const SizeType NumKnotSpansV,
        const bool AddSurfaceToModelPart)
```
(bear the style guide...)